### PR TITLE
cleaned and updated Mafft controller

### PR DESCRIPTION
**Pull request by RNAer**

### CODE REVIEW

**Comment by ekopylova** [comment]
@RNAer, do you remember the specific changes you've made that differentiate the earlier version of the app controller for Mafft?

**Comment by RNAer** [comment]
I recall it includes a upgrade for a new Mafft version and a general
cleanup

On Fri, Jul 24, 2015 at 1:11 PM, Evguenia Kopylova <notifications@github.com

File: bfillings/mafft_v7.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+        else:
+            raise ValueError("No tree output file specified.")
+        return tree_filename
+
+    def getHelp(self):
+        """Method that points to the Mafft documentation."""
+
+        help_str = ("See Mafft documentation at:\n"
+                    "http://align.bmr.kyushu-u.ac.jp/"
+                    "mafft/software/manual/manual.html")
```
Comment (ekopylova) [inline bfillings/mafft_v7.py]: I'm getting page not found for this link ..

File: bfillings/mafft_v7.py [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+    seqs_fp : string
+        file path of the input fasta file
+    moltype : callable constructor
+        It can be ``DNA``, ``RNA``, ``Protein`` classes that are defined in
+        ``skbio``. Default is DNA
+    params : dict-like type
+        It pass the additional parameter settings to the application.
+        Default is None.
+    accurate : boolean
+        Perform accurate alignment or not. It will sarificing performance
```
Comment (ekopylova) [inline bfillings/mafft_v7.py]: sacrifice

File: bfillings/mafft_v7.py [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+    sequences.
+
+    Parameters
+    ----------
+    seqs_fp : file path of the unaligned sequences
+
+    aln_fp : file path of the seed alignment
+
+    params : dict of parameters to pass in to the Mafft app controller.
+
```
Comment (ekopylova) [inline bfillings/mafft_v7.py]: remove space between parameter declarations, add definition for parameters moltype and accurate

File: bfillings/mafft_v7.py [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+    """Returns an Alignment object from two existing Alignments.
+
+    Parameters
+    ----------
+    aln1_fp : file path of 1st alignment
+
+    aln2_fp : file path of 2nd alignment
+
+    params : dict of parameters to pass in to the Mafft app controller.
+
```
Comment (ekopylova) [inline bfillings/mafft_v7.py]: same as above (spaces & parameter declaration)

File: bfillings/tests/test_mafft_v7.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+                                 ">a\n"
+                                 "-------uaggcucugauauaauagcucuc---\n"
+                                 ">b\n"
+                                 "uaucgcuucgacgauucucugauagaga-----\n"
+                                 ">c\n"
+                                 "-------------------ugacuacgcau---\n")
+        # self.aligned2 = {'a': 'UAGGCUCUGAUAUAAUAGCUCUC---------',
+        #                  'b': 'UA----UCGCUUCGACGAUUCUCUGAUAGAGA',
+        #                  'c': 'UG------------ACUACGCAU---------'}
+
```
Comment (ekopylova) [inline bfillings/tests/test_mafft_v7.py]: if self.aligned2 is not needed, remove?

**Comment by ekopylova** [comment]
I'm getting only 69% test coverage for the code, is it possible to improve this?
bfillings/mafft_v7         52     16    69%   217-222, 225-229, 234-237, 242-243, 287-288, 326
bfillings/tests/test_mafft_v7       54      0   100%

**Comment by ekopylova** [comment]
@josenavas could you review this as well? thanks!

File: bfillings/mafft_v7.py [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+
+def align_unaligned_seqs(seqs_fp, moltype=DNA, params=None, accurate=False):
+    """Aligns unaligned sequences
+
+    Parameters
+    ----------
+    seqs_fp : string
+        file path of the input fasta file
+    moltype : callable constructor
```
Comment (josenavas) [inline bfillings/mafft_v7.py]: `moltype : {skbio.DNA, skbio.RNA, skbio.Protein}`
This is the way lists of options are listed in numpy documentation.

**Comment by RNAer** [comment]
updated

**Comment by ekopylova** [comment]
@gregcaporaso can you look this over and merge? We need this code for the deblur pipeline. Thanks!

**Comment by gregcaporaso** [comment]
This looks good on a quick pass through. What are the big differences from the previous version of Mafft?

**Comment by RNAer** [comment]
There are quite some changes: http://mafft.cbrc.jp/alignment/software/changelog.html

**Comment by gregcaporaso** [comment]
Merged, thanks!

**Pull request merged**